protocol/x/clob/keeper/process_operations.go

```diff
@@ -151,6 +145,19 @@ func (k Keeper) ProcessInternalOperations(
 
 		switch castedOperation := operation.Operation.(type) {
 		case *types.InternalOperation_Match:
+			// check if affiliate whitelist map is nil and initialize it if it is.
+			// This is done to avoid getting whitelist map on list of operations
+			// where there are no matches.
+			if affiliatesWhitelistMap == nil {
```

Comment:
If the `AffiliateWhitelist` is empty in state, then this will be `nil` everytime right? Do we need a separate boolean to track whether the state read was done?

Reply:
golang dfiferentiates between a nil map and an empty map

Reply:
```
package main

import "fmt"

func main() {
	fmt.Println("Hello, 世界")
	x := map[string]uint32{}
	if x == nil {
		fmt.Println("x is nil")
	}
	var y map[string]uint32
	if y == nil {
		fmt.Println("y is nil")
	}
}

```

Reply:
try this piece of code(confirmed it myself)